src/protocol/resp/src/request/incr.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use super::*;
+use logger::klog;
+use std::io::{Error, ErrorKind};
+use std::sync::Arc;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Incr {
+    key: Arc<[u8]>,
+}
+
+impl TryFrom<Message> for Incr {
+    type Error = Error;
+
+    fn try_from(other: Message) -> Result<Self, Error> {
+        if let Message::Array(array) = other {
+            if array.inner.is_none() {
+                return Err(Error::new(ErrorKind::Other, "malformed command"));
+            }
+
+            let mut array = array.inner.unwrap();
+
+            if array.len() != 2 {
+                return Err(Error::new(ErrorKind::Other, "malformed command"));
+            }
+
+            let key = if let Message::BulkString(key) = array.remove(1) {
+                if key.inner.is_none() {
+                    return Err(Error::new(ErrorKind::Other, "malformed command"));
+                }
+
+                let key = key.inner.unwrap();
+
+                if key.len() == 0 {
+                    return Err(Error::new(ErrorKind::Other, "malformed command"));
+                }
+
+                key
+            } else {
+                return Err(Error::new(ErrorKind::Other, "malformed command"));
+            };
+
+            Ok(Self { key })
+        } else {
+            Err(Error::new(ErrorKind::Other, "malformed command"))
+        }
+    }
+}
+
+impl Incr {
+    pub fn new(key: &[u8]) -> Self {
+        Self { key: key.into() }
+    }
+
+    pub fn key(&self) -> &[u8] {
+        &self.key
+    }
+}
+
+impl From<&Incr> for Message {
+    fn from(other: &Incr) -> Message {
+        Message::Array(Array {
+            inner: Some(vec![
+                Message::BulkString(BulkString::new(b"INCR")),
+                Message::BulkString(BulkString::from(other.key.clone())),
+            ]),
+        })
+    }
+}
+
+impl Compose for Incr {
+    fn compose(&self, buf: &mut dyn BufMut) -> usize {
+        let message = Message::from(self);
+        message.compose(buf)
+    }
+}
+
+impl Klog for Incr {
```

Comment:
TODO: I have to verify that this implementation is as close to memcache incr as possible